## QIIME2 metadata
```{r}
library(tidyverse, suppressPackageStartupMessages())

# QIIME2  metadata
read_tsv("../../../imap-bioinformatics-qiime2/resources/metadata/qiime2_sample_metadata.tsv", show_col_types = FALSE) %>% 
  rename(sample_id="sample-id") %>% 
  write_csv("../../data/qiime2/qiime2_tidy_metadata.csv")
```


## QIIME2  otutable
```{r}
read_tsv("../../../imap-bioinformatics-qiime2/qiime2_process/export/feature-table.tsv", skip = 1, show_col_types = FALSE) %>%
  rename(feature='#OTU ID') %>%
  select(-starts_with('Mock')) %>% 
  mutate_at(2:ncol(.), as.numeric) %>% 
  mutate_if(is.numeric, ~replace(., is.na(.), 0)) %>% 
  pivot_longer(-feature, names_to = "sample_id", values_to = "count") %>% 
  relocate(sample_id, .before = feature) %>% 
  write_csv("../../data/qiime2/qiime2_tidy_otutable.csv")
```


## QIIME2 taxonomy
```{r}
read_tsv("../../../imap-bioinformatics-qiime2/qiime2_process/export/taxonomy.tsv", show_col_types=FALSE) %>% 
  rename(feature="Feature ID") %>% 
  distinct() %>%
  mutate_if(is.numeric, ~replace(., is.na(.), 0)) %>%
  mutate(Taxon = str_replace_all(Taxon, "; s__$", ""),
         Taxon = str_replace_all(Taxon, "; g__$", ""),
         Taxon = str_replace_all(Taxon, "; f__$", ""),
         Taxon = str_replace_all(Taxon, "; o__$", ""),
         Taxon = str_replace_all(Taxon, "; c__$", ""),
         Taxon = str_replace_all(Taxon, "; p__$", ""),
         Taxon = str_replace_all(Taxon, "; k__$", ""),
         Taxon = str_replace_all(Taxon, "\\[|\\]", ""),
         Taxon = str_replace_all(Taxon, "\\s", "")) %>%
  dplyr::filter(!grepl("s__*", Taxon)) %>%
  dplyr::filter(grepl("g__*", Taxon)) %>% 
  select(-Confidence) %>% 
  mutate(Taxon = str_replace_all(Taxon, "\\w__", "")) %>% 
  separate(Taxon, into = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus"), sep = ";") %>% 
  write_csv("../../data/qiime2/qiime2_tidy_taxonomy.csv")

```


## QIIME2 composite
```{r}
# QIIME2 composite
library(tidyverse, suppressPackageStartupMessages())

qiime2_tidy_metadata <- read_csv("../../data/qiime2/qiime2_tidy_metadata.csv", show_col_types = FALSE)
qiime2_tidy_otutable <- read_csv("../../data/qiime2/qiime2_tidy_otutable.csv", show_col_types = FALSE)
qiime2_tidy_taxonomy <- read_csv("../../data/qiime2/qiime2_tidy_taxonomy.csv", show_col_types = FALSE)

qiime2_composite <- inner_join(qiime2_tidy_metadata, qiime2_tidy_otutable, by = "sample_id") %>% 
  inner_join(., qiime2_tidy_taxonomy, by = "feature") %>% 
  group_by(sample_id) %>% 
  mutate(rel_abund = count/sum(count)) %>% 
  ungroup() %>% 
  mutate_if(is.numeric, ~replace(., is.na(.), 0)) %>%
  relocate(count, .before = rel_abund) 

write_csv(qiime2_composite, "../../data/qiime2/qiime2_composite.csv")
```

```{r}
library(tidyverse)
library(phyloseq)
library(microbiome)
library(ape)

metadata <- read_csv("../../data/qiime2/qiime2_tidy_metadata.csv",show_col_types = FALSE) %>% 
  tibble::column_to_rownames("sample_id") %>% 
  sample_data(metadata)

otutable <- read_csv("../../data/qiime2/qiime2_tidy_features.csv",show_col_types = FALSE) %>% 
  pivot_wider(id_cols = sample_id, names_from = feature, values_from = count) %>% 
  tibble::column_to_rownames("sample_id") %>% 
  otu_table(otutable, taxa_are_rows = FALSE)

taxonomy <- read_csv("../../data/qiime2/qiime2_tidy_taxonomy.csv", show_col_types = FALSE) %>%
  tibble::column_to_rownames("feature") %>%
  as.matrix() %>% 
  tax_table(taxonomy)

ps_raw_basic <- merge_phyloseq(metadata, otutable, taxonomy)

library(ape)
ps_tree = rtree(ntaxa(ps_raw_basic), rooted=TRUE, tip.label=taxa_names(ps_raw_basic))
ps_raw <- phyloseq::merge_phyloseq(ps_raw_basic, ps_tree)

ps_rel <- phyloseq::transform_sample_counts(ps_raw, function(x){x / sum(x)})

save(ps_tree, ps_raw, ps_rel,  file = "data/qiime2/qiime2_phyloseq_objects.rda")
```

